file.c

```diff
@@ -1621,11 +1621,15 @@ rb_file_directory_p(VALUE obj, VALUE fname)
 
 /*
  * call-seq:
- *   File.pipe?(file_name)   ->  true or false
+ *   File.pipe?(object) -> true or false
  *
- * Returns <code>true</code> if the named file is a pipe.
+ * Returns +true+ if +object+ is a pipe, +false+ otherwise:
```

Comment:
The main use of `File.pipe?` is with file names, though it does also support IO objects.  There should be a description and examples for use with file names (pipes in the file system can be created with `File.mkfifo`).

Similar issue for `File.socket?`